src/AddressRegistry.Api.BackOffice.Handlers.Lambda/Handlers/ReaddressLambdaHandler.cs

```diff
@@ -0,0 +1,124 @@
+            ICustomRetryPolicy retryPolicy,
+            ITicketing ticketing,
+            IStreetNames streetNames,
+            IIdempotentCommandHandler idempotentCommandHandler,
+            BackOfficeContext backOfficeContext)
+            : base(
+                configuration,
+                retryPolicy,
+                streetNames,
+                ticketing,
+                idempotentCommandHandler)
+        {
+            _backOfficeContext = backOfficeContext;
+        }
+
+        protected override async Task<object> InnerHandle(ReaddressLambdaRequest request, CancellationToken cancellationToken)
+        {
+            // Get streetNamePersistentLocalId for each address in the request
+            // TODO: Describe steps being executed
+
+            var readdressAddressItems = new List<ReaddressAddressItem>();
+
+            foreach (var item in request.Request.HerAdresseer)
+            {
+                var addressPersistentLocalId =
+                    new AddressPersistentLocalId(Convert.ToInt32(item.BronAdresId.AsIdentifier().Map(x => x).Value));
+                var relation = await _backOfficeContext.FindRelationAsync(addressPersistentLocalId, cancellationToken);
+
+                var readdressAddressItem = new ReaddressAddressItem(
+                    new StreetNamePersistentLocalId(relation.StreetNamePersistentLocalId),
+                    addressPersistentLocalId,
+                    HouseNumber.Create(item.DoelHuisnummer));
+
+                readdressAddressItems.Add(readdressAddressItem);
+            }
+
+            var retireAddressItems = new List<RetireAddressItem>();
+
+            foreach (var item in request.Request.OpheffenAdressen ?? new List<string>())
+            {
+                var addressPersistentLocalId =
+                    new AddressPersistentLocalId(Convert.ToInt32(item.AsIdentifier().Map(x => x).Value));
+
+                // We don't look into the context because OpheffenAdressen should all be in HerAdresseer
+                var streetNamePersistentLocalId = readdressAddressItems
+                        .Single(x => x.SourceAddressPersistentLocalId == addressPersistentLocalId)
+                        .SourceStreetNamePersistentLocalId;
+
+                retireAddressItems.Add(new RetireAddressItem(streetNamePersistentLocalId, addressPersistentLocalId));
+            }
+
+            var cmd = request.ToCommand(readdressAddressItems, retireAddressItems);
+
+            try
+            {
+                // pass context object by reference
+                await IdempotentCommandHandler.Dispatch(
+                    cmd.CreateCommandId(),
+                    cmd,
+                    request.Metadata,
+                    cancellationToken);
+            }
+            catch (IdempotencyException)
+            {
+                // Idempotent: Do Nothing return last etag
+            }
+            // TODO: ExecutionContext is empty when idempotencyException was thrown
```

Comment:
@ArneD given the scenario that the domain logic succeeds and its events are persisted, but then an error would occur a bit further in the lambda handler (e.g. an action on the BackOfficeContext), then we'd perform a retry. In this retry,  an idempotency exception would be thrown leaving the ExecutionContext empty. With an empty ExecutionContext we can't return a proper list of etag responses. Do we add this problem as a task to the story or do we want to tackle this issue in this PR?

Reply:
I prefer now. There are a few ways we can do this.
- Get events by correlation id ( command id)
- Get addresses by the data given in the request
- ... ?